Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -3944,18 +3958,62 @@ struct RenderBox::PositionedLayoutConstraints {
         marginPercentageBasis = containingWidth;
 
         computeAnchorGeometry(self);
-    }
 
+        captureInsets(self, logicalAxis);
+    }
+    void captureInsets(const RenderBox& self, const LogicalBoxAxis selfAxis);
     void computeAnchorGeometry(const RenderBox& self);
     LayoutRange adjustForPositionArea(const LayoutRange rangeToAdjust, const LayoutRange anchorArea, const BoxAxis containerAxis, const RenderStyle&);
 
     void computeInlineStaticDistance(const RenderBox& self);
     void computeBlockStaticDistance(const RenderBox& self);
 
     void convertLogicalLeftValue(LayoutUnit& logicalLeftValue, const RenderBox* child, const LayoutUnit logicalWidthValue, const bool logicalLeftIsAuto, const bool logicalRightIsAuto) const;
-    void convertLogicalTopValue(LayoutUnit& logicalTopValue, const RenderBox* child, const LayoutUnit logicalHeightValue, const bool logicalTopIsAuto, const bool logicalBottomIsAuto) const;
+    void convertLogicalTopValue(LayoutUnit& logicalTopValue, const RenderBox& self, const LayoutUnit logicalHeightValue) const;
+
+    void resolvePosition(LogicalExtentComputedValues&) const;
 };
 
+void RenderBox::PositionedLayoutConstraints::captureInsets(const RenderBox& self, const LogicalBoxAxis selfAxis)
```

Comment:
I'd refrain from landing more changes like this (`const LogicalBoxAxis selfAxis`) until it has been discussed on WebKit.